src/js_app.hpp

```diff
@@ -355,5 +382,46 @@ void AppClass<T>::set_versions(ContextType ctx, ObjectType this_object, Argument
     AppClass<T>::platform_version = Object::validated_get_string(ctx, versions, "platformVersion");
 }
 
+template <typename T>
+void AppClass<T>::add_listener(ContextType ctx, ObjectType this_object, Arguments& args, ReturnValue& return_value)
+{
+    auto callback = Value::validated_to_function(ctx, args[0]);
+    auto app = get_internal<T, AppClass<T>>(ctx, this_object);
+    Protected<FunctionType> protected_callback(ctx, callback);
+    Protected<ObjectType> protected_this(ctx, this_object);
+    Protected<typename T::GlobalContext> protected_ctx(Context::get_global_context(ctx));
+
+    {
+        auto token = std::move(app->m_app->subscribe([=](const realm::app::App&) {
+            Function::callback(protected_ctx, protected_callback, 0, {});
+        }));
+
+        // Save token in a member vector of a function to token pair
+        app->m_notification_tokens.emplace_front(std::move(protected_callback), std::move(token));
+    }
```

Comment:
I might be misunderstanding something here but does block scoping this serve any purpose?

```suggestion
        auto token = std::move(app->m_app->subscribe([=](const realm::app::App&) {
            Function::callback(protected_ctx, protected_callback, 0, {});
        }));

        // Save token in a member vector of a function to token pair
        app->m_notification_tokens.emplace_front(std::move(protected_callback), std::move(token));
```

Reply:
It was a left over from me trying to understand why the Token was being destructed and unsubscribing itself when the temp value went out of scope.